prefect_dask/task_runners.py

```diff
@@ -308,9 +314,8 @@ def visit_fn(expr):
     async def wait(self, key: UUID, timeout: float = None) -> Optional[State]:
         future = self._get_dask_future(key)
         try:
-            return await future.result(timeout=timeout)
-        except distributed.TimeoutError:
-            return None
+            async for _, res in future:
+                return res
```

Comment:
I imagine you can still enforce a timeout here using `with anyio.move_on_after(timeout)`